### ISSUE TITLE
Island labels given higher priority than the water bodies they reside in

### ISSUE THREAD

**Issue by MrAce100** [organicmaps/organicmaps#9055]
**Describe the issue**
At certain zoom levels, labels for islands appear when the body of water they are in is not yet visible, cluttering up the map.

**Steps to reproduce**
1. Search for "Zanesville Ohio"
2. Click on the result for the US city at 676 ft elevation.

**Expected behaviour**
Island labels appear at the appropriate zoom level, after the body of water becomes visible.

**Screenshots**
![Screenshot_20240822-105353_Organic Maps](https://github.com/user-attachments/assets/912bb204-69f0-4642-a64a-c31f3c8f619e)

**System information:**
 - Operating system and its version: Android 14 (Lineage OS)
 - Organic Maps version: 2024.08.16-5-FDroid
 - Device Model: Google Pixel 3a

Thanks :)

**Comment by Ely31** [comment]
Yes, this is very annoying. Sometimes it renders the names of tiny islands before entire cities.
Here it renders Canary island before Menominee
![canary island far out](https://github.com/user-attachments/assets/e066d79e-5ba2-458a-93d8-40d900161b30)
And this is the size of that island
![canary island closeup](https://github.com/user-attachments/assets/64a04033-c420-4c44-a66b-21930f1404dd)

**Comment by brdbkr** [comment]
It's especially absurd in coastal areas:

![IMG_1112](https://github.com/user-attachments/assets/20bcc8f0-1da1-4953-b032-ef0ae9b843c4)
![IMG_1113](https://github.com/user-attachments/assets/8433d5ca-d7e6-4859-b947-896744ba84b7)

**Comment by pastk** [comment]
Probably we should provide different styling for islands' captions based on their size/area (bbox).
E.g. divide into `place-island-big`, `place-island-med`, `place-island-small` feature types at the generation stage (and there is a separate `place-islet` already).
